Lib/unittest/mock.py

```diff
@@ -1589,7 +1589,7 @@ def stop(self):
 def _get_target(target):
     try:
         target, attribute = target.rsplit('.', 1)
-    except (TypeError, ValueError):
+    except (TypeError, ValueError, AttributeError):
         raise TypeError("Need a valid target to patch. You supplied: %r" %
                         (target,))
```

Comment:
```suggestion
        raise TypeError(f"Need a valid target to patch. You supplied: {target}")
```

since there is a change of nearby lines anyway. Plus, "b'%r' is deprecated, but will not be removed during the 3.x series" [in the docs](https://docs.python.org/3/library/stdtypes.html).

Reply:
Should be `{target!r}` I suppose?

Reply:
Ah, yes. I was optimistic seeing string-only nature of `target`:

https://github.com/python/cpython/blob/461e59e4f4ac839c2bbdecf478cca1fec93068a9/Lib/unittest/mock.py#L1693

Like, once it was an object requiring some special formatting but now it's just a string. However, I forgot that `%r` additionally escapes non-ascii code points.

Reply:
Right, also this bpo is about the case where the error is that you passed in some non-string object.